lxd-migrate/main_migrate.go

```diff
@@ -556,6 +597,16 @@ func (c *cmdMigrate) run(cmd *cobra.Command, args []string) error {
 		if err != nil {
 			return fmt.Errorf("Failed to make %s read-only: %w", config.SourcePath, err)
 		}
+
+		// In conversion mode, server expects the volume size hint in the request.
+		if config.InstanceArgs.Source.Type == "conversion" {
+			stat, err := os.Stat(target)
+			if err != nil {
+				return err
+			}
+
+			config.InstanceArgs.Source.VolumeSize = stat.Size()
```

Comment:
Should we call `VolumeSize`, just `Size` and then its correct for both raw and compressed inputs - an its up to the receiver to treat it either as actual volume size or not depending on whether converting?

Also, out of interest, the instance device config is set in the request right, so can we have lxd-migrate just update the `root` disk's `size` property, rather than needing a separate field anyway?

Reply:
I'll investigate this one and report back

Reply:
> Also, out of interest, the instance device config is set in the request right, so can we have lxd-migrate just update the root disk's size property, rather than needing a separate field anyway?

The root device config is set if the user specifies the storage pool. 
Otherwise, we do not know what storage pool will be used, and the root device is not set (the server then determines the storage pool based on the project and applied profiles)